app/controllers/contacts_controller.rb

```diff
@@ -13,13 +13,17 @@ def index
 
       # Used by the mobile site
       format.mobile { set_up_contacts_mobile }
+    end
+  end
+
+  def search
+    @people = Person.contacts_search(params[:q], current_user)
 
-      # Used to populate mentions in the publisher
-      format.json {
-        aspect_ids = params[:aspect_ids] || current_user.aspects.map(&:id)
-        @people = Person.all_from_aspects(aspect_ids, current_user).for_json
-        render :json => @people.to_json
-      }
+    respond_to do |format|
+      format.json do
+        @people = @people.limit(15)
```

Comment:
Does the rails limit works like the SQL limit (meaning it only truncates result at the end but the load is still heavy?)